### ISSUE TITLE
Add `transaction.name` into errors and spans

### ISSUE THREAD

**Issue by ErvalhouS** [elastic/apm-agent-ruby#1342]
I have an external visualization tool that lacks the ability to group spans/errors into application actions, without relying on a costly `transaction.id` join style listing.

If the spans and errors had a `transaction.name` metadata on it ([which should be there as per documentation](https://www.elastic.co/guide/en/apm/server/current/exported-fields-apm-error.html#_service)), that operation would make way less pressure on the Elasticsearch cluster.

**Comment by smith** [comment]
I don't think we do this anywhere. Those docs don't explicitly state those attributes are on spans and I don't think they usually are.
